storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -330,7 +330,11 @@ public static Map readCommandLineOpts() {
         Map ret = new HashMap();
         String commandOptions = System.getProperty("storm.options");
         if (commandOptions != null) {
-            String[] configs = commandOptions.split(",");
+            /*
+             below regex uses negative lookahead to not split in the middle of
+             json objects '{}' or json arrays '[]'
+              */
+            String[] configs = commandOptions.split(",(?![^\\[\\]{}]*(]|}))");
```

Comment:
Sorry about the long pause.  I thought I had replied already, but I guess I didn't hit the merge comment button.  I forgot that the python was escaping all of the JSON characters, so like you said this works.

I am OK with merging this in, but I would like an explanation by the REGEX explaining about the escaping in most cases and that this is not perfect for windows.

It would also be good to have a follow on JIRA to solve this correctly with url encoding the parameters, but I don't know enough about windows to suggest how to do that.